src/hub.js

```diff
@@ -908,19 +942,26 @@ document.addEventListener("DOMContentLoaded", async () => {
             scene.removeState("copresent");
           }
 
-          if (!isInitialSync) return;
-          // Wire up join/leave event handlers after initial sync.
-          isInitialSync = false;
+          // HACK - Set a flag on the presence object indicating if the initial sync has completed,
+          // which is used to determine if we should fire join/leave messages into the presence log.
+          presence.__hadInitialSync = true;
+
+          presence.onJoin((sessionId, current, info) => {
+            if (!hubChannel.presence.__hadInitialSync) return;
```

Comment:
Might be worth adding a comment here that these onSync, onJoin, and onLeave handlers are reused when migrating servers, since this early exit doesn't make sense if you don't know about that context.

Reply:
Specifically that this early exit should only ever happen if we are in the middle of a migration (if I understand correctly)

Reply:
the early exit actually happens for the first sync event for each `Presence` object -- typically just the one we create at startup. but if you reconnect, we create another, and skip onJoins as well here in that scenario (so you don't see the whole room join when you reconnect.) i'll update the comment

Reply:
I don't think I understand how the early exit applies on the very first sync, given we set `__hadInitialSync` to true immediately before registering this handler.